src/protocols/fp-meta.ts

```diff
@@ -0,0 +1,137 @@
+import { FPEnvelopeTurnaround, FPEnvelopeType, FPStoreType } from "./fp-envelope";
+import { CID } from "multiformats";
+
+export interface FPGetMetaReq {
+    readonly cid: CID;
+}
+/*
+ * A request to get a META file.
+ * As JSON, this is:
+ * {
+ *    "type": "get-meta-req",
+ *    "tid": "123",
+ *    "payload": {
+ *       "cid": "bafybeib2v"
+ *    }
+ *    "auth": { ... }
+ *    "meta": { ... }
+ * }
+ */
+export interface FPEnvelopeGetMetaReq extends FPMetaReq<FPGetMetaReq> {
+    readonly type: FPEnvelopeType.GET_METAREQ;
+    readonly storeType: FPStoreType.META;
+}
+
+export interface FPGetMetaRes {
+    readonly cid: CID;
+    readonly bytes: Uint8Array;
+}
+/*
+ * The response to a GetMeta request.
+ * As JSON, this is:
+ * {
+ *    "type": "get-meta-res",
+ *    "tid": "123",
+ *    "payload": {
+ *       "cid": "bafybeib2v"
+ *       "bytes": [0x65, 0x66, 0x67, ...],
+ *    }
+ *    "auth": { ... }
+ *    "meta": { ... }
+ * }
+ */
+export interface FPEnvelopeGetMetaRes extends FPMetaRes<FPGetMetaRes> {
+    readonly type: FPEnvelopeType.GET_METARES;
+    readonly storeType: FPStoreType.META;
+}
+
+export interface FPPutMetaReq {
+    readonly cid: CID;
+    readonly bytes: Uint8Array;
+}
+/*
+ * A request to get a META file.
+ * As JSON, this is:
+ * {
+ *    "type": "put-meta-req",
+ *    "tid": "123",
+ *    "payload": {
+ *       "cid": "bafybeib2v"
+ *       "bytes": [0x65, 0x66, 0x67, ...],
+ *    }
+ *    "auth": { ... }
+ *    "meta": { ... }
+ * }
+ */
+export interface FPEnvelopePutMetaReq extends FPMetaReq<FPPutMetaReq> {
+    readonly type: FPEnvelopeType.PUT_METAREQ;
+    readonly storeType: FPStoreType.META;
+}
+
+export interface FPDelMetaReq {
+    readonly cid: CID;
+}
+/*
+ * The response to a PutMeta request.
```

Comment:
this should respond with the HEAD post-PUT, inclusive of the PUTs effect -- same as a GET.